docs/resources/branding_theme.md

```diff
@@ -11,6 +11,23 @@ This resource allows you to manage branding themes for your Universal Login page
 ## Example Usage
 
 ```terraform
+# An example of the auth0_branding_theme using defaults for all the attributes.
+resource "auth0_branding_theme" "my_theme" {
+  borders {}
+  colors {}
+  fonts {
+    title {}
+    subtitle {}
+    links {}
+    input_labels {}
+    buttons_text {}
+    body_text {}
+  }
+  page_background {}
+  widget {}
```

Comment:
There are a couple of problems and challenges to allow setting the minimal configuration for this resource to:

```hcl
resource "auth0_branding_theme" "my_theme" { }
```

1. If we set all the `TypeList` elements to `Optional` we can't ensure that the `MinItems` count is 1 because that is ignored when `Optional` is true.
2. We can't use the Default property on TypeList or TypeSets. From the docs:
```hcl
  // Default cannot be used if the Schema is directly an
	// implementation of an Elem field of another Schema, such as trying to
	// set a default value for a TypeList or TypeSet.
```

To work around this I've noticed that your attempt @nialdaly was to handle this within the `expandBrandingTheme` func, however that opens us up for serious drifts between the default values we set on the schema and potential erroneous changes of the default values within the expand func.

So to keep just the defaults on the schema as the only sources of truth we're requiring all the TypeList blocks to be set within the config at least as empty. The example highlighted here in the code above.

Reply:
@sergiught my initial approach was to require all of the blocks and then I went down the rabbit hole of trying to achieve the minimal configuration. I'm not an active user but I think the approach you suggest makes sense as the API does actually require the values!